packages/mds-ingest-service/service/provider.ts

```diff
@@ -103,6 +106,25 @@ export const IngestServiceProvider: ServiceProvider<IngestService & IngestMigrat
     }
   },
 
+  writeEvents: async (events: EventDomainCreateModel[]) => {
+    try {
+      // Check to see if the devices references in the events exist in the db. This is in a closure to keep the code pretty boxed off, but I didn't feel the need to make a named function elsewhere :)
+      // FIXME: Move this check to the cache (once we have a cache manged by the ingest service)
+      await (async () => {
+        const devicesInPayload = new Set(events.map(e => e.device_id))
```

Comment:
I have a question that's a bit out of scope for this PR, but what is the client expected to do if not all the devices are in the DB? Is it going to register the devices and attempt to write the events again?

Reply:
Yup! Worth noting that in reality, at least for the foreseeable future, this will only be used for writing one event at a time.